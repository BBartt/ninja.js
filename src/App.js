import React from 'react';
import DataTable from './DataTable';

import { dummyData } from './dummyData/dummyData';

import './scss/app.scss';

function App() {
  return (
    <div className="container mt-3">
      <DataTable data={dummyData} rowsPerPage={5} />
    </div>
  );
}

export default App;
