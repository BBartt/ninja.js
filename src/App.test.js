import React from 'react';

import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import App from './App';

import { dummyData } from './dummyData/dummyData';

describe('<App />', () => {
  it('renders without crashing', () => {
    shallow(<App rows={[]} rowsPerPage={5} />);
  });

  it('matches snapshot', () => {
    const tree = renderer.create(<App rows={[]} rowsPerPage={5} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders 5 rows', () => {
    const wrapper = mount(<App rows={dummyData} rowsPerPage={5} />);

    expect(wrapper.find('tr').length).toBe(5);
  });

  it('filters rows based on input', () => {
    const wrapper = mount(<App rows={dummyData} rowsPerPage={5} />);

    wrapper.find('input').simulate('change', { target: { value: '.us' } });

    expect(wrapper.find('tr').length).toBe(1);
  });

  it('display "no data" row when no record', () => {
    const wrapper = mount(<App rows={dummyData} rowsPerPage={5} />);

    wrapper
      .find('input')
      .simulate('change', { target: { value: 'lorem ipsum dolor' } });

    expect(wrapper.find('td').text()).toContain('No data');
    expect(wrapper.find('tr').length).toBe(1);
  });
});
