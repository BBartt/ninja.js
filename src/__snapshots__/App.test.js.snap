// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches snapshot 1`] = `
<div
  className="container mt-3"
>
  <div
    className="dataTable"
  >
    <div
      className="p-b-1"
    >
      <input
        className="form-control"
        onChange={[Function]}
        placeholder="SÃ¸g brugere"
        type="search"
        value=""
      />
    </div>
    <table>
      <tbody>
        <tr>
          <td>
            <a
              href="http://google.com"
            >
              Mads L. Klausen
            </a>
            <br />
            <small>
              MadsLKlausen@jourrapide.com
            </small>
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="http://google.com"
            >
              Alfred K. Krogh
            </a>
            <br />
            <small>
              AlfredKKrogh@armyspy.com
            </small>
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="http://google.com"
            >
              Silas L. Bertelsen
            </a>
            <br />
            <small>
              SilasLBertelsen@armyspy.com
            </small>
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="http://google.com"
            >
              Mia A. Johnsen
            </a>
            <br />
            <small>
              MiaAJohnsen@dayrep.com
            </small>
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="http://google.com"
            >
              Alfred S. Schou
            </a>
            <br />
            <small>
              AlfredSSchou@jourrapide.com
            </small>
          </td>
        </tr>
      </tbody>
    </table>
    <ul
      className="pagination"
    >
      <li
        className="page-item mr-1"
      >
        <button
          className="page-link"
          onClick={[Function]}
        >
          1
        </button>
      </li>
      <li
        className="page-item mr-1"
      >
        <button
          className="page-link button-outline"
          onClick={[Function]}
        >
          2
        </button>
      </li>
      <li
        className="page-item mr-1"
      >
        <button
          className="page-link"
          onClick={[Function]}
        >
          3
        </button>
      </li>
      <li
        className="page-item mr-1"
      >
        <button
          className="page-link"
          onClick={[Function]}
        >
          4
        </button>
      </li>
      <li
        className="page-item mr-1"
      >
        <button
          className="page-link"
          onClick={[Function]}
        >
          5
        </button>
      </li>
      <li
        className="page-item mr-1"
      >
        <button
          className="page-link"
          onClick={[Function]}
        >
          6
        </button>
      </li>
      <li
        className="page-item mr-1"
      >
        <button
          className="page-link"
          onClick={[Function]}
        >
          7
        </button>
      </li>
      <li
        className="page-item mr-1"
      >
        <button
          className="page-link"
          onClick={[Function]}
        >
          8
        </button>
      </li>
      <li
        className="page-item mr-1"
      >
        <button
          className="page-link"
          onClick={[Function]}
        >
          9
        </button>
      </li>
      <li
        className="page-item mr-1"
      >
        <button
          className="page-link"
          onClick={[Function]}
        >
          10
        </button>
      </li>
    </ul>
  </div>
</div>
`;
